### PR TITLE
IPv6 support for m_dnsbl and sockaddrs::reverse()

### PR DESCRIPTION
**Pull request by key2peace**
Moved m_dns's reverse ip method to sockaddrs::reverse().
(Since m_dnsbl also relies on it, I thought it would be easier to move it to there where it belongs in order to save duplicate code on something generic)
Updated m_dnsbl to support ipv6.

### CODE REVIEW

File: modules/m_dnsbl.cpp [hunk 1]
```diff
@@ -174,17 +173,11 @@ class ModuleDNSBL : public Module
...
-
-		sockaddrs reverse = user->ip;
-		reverse.sa4.sin_addr.s_addr = reverse_ip;
-
 		for (unsigned i = 0; i < this->blacklists.size(); ++i)
 		{
 			const Blacklist &b = this->blacklists[i];
 
-			Anope::string dnsbl_host = reverse.addr() + "." + b.name;
+			Anope::string dnsbl_host = user->ip.reverse() + "." + b.name;
```
Comment (Adam-) [inline modules/m_dnsbl.cpp]: Only call reverse() once and store the result instead of for each blacklist

File: modules/m_dns.cpp [hunk 2]
```diff
@@ -295,33 +295,18 @@ class Packet : public Query
...
-					q.name = ip.addr() + ".in-addr.arpa";
+					case AF_INET6:
+						q.name = ip.reverse() + ".ip6.arpa";
+						break;
+					case AF_INET:
+						q.name = ip.reverse() + ".in-addr.arpa";
+						break;
+					default:
+						throw SocketException("Unsupported IP Family");
+						break;
```
Comment (Adam-) [inline modules/m_dns.cpp]: It is unnecessary to break; after throwing an exception

File: src/sockets.cpp [hunk 3]
```diff
@@ -95,6 +95,43 @@ Anope::string sockaddrs::addr() const
...
 
+Anope::string sockaddrs::reverse() const
+{
+	char address[128];
+
+	switch (sa.sa_family)
+	{
+		case AF_INET6:
+		{
+			const char *const hex = "0123456789abcdef";
```
Comment (Adam-) [inline src/sockets.cpp]: Make this static const char[] hex = "0123456789abcdef";

File: modules/m_dnsbl.cpp [hunk 4]
```diff
@@ -174,17 +173,13 @@ class ModuleDNSBL : public Module
 			return;
 		}
 
-		const unsigned long &ip = user->ip.sa4.sin_addr.s_addr;
-		unsigned long reverse_ip = (ip << 24) | ((ip & 0xFF00) << 8) | ((ip & 0xFF0000) >> 8) | (ip >> 24);
-
-		sockaddrs reverse = user->ip;
-		reverse.sa4.sin_addr.s_addr = reverse_ip;
-
+		Anope::string reverse =  user->ip.reverse();
```
Comment (Adam-) [inline modules/m_dnsbl.cpp]: There is a double space here

File: modules/m_dns.cpp [hunk 5]
```diff
@@ -295,33 +295,17 @@ class Packet : public Query
...
-						reverse_ip[reverse_ip_count++] = hex[ip.sa6.sin6_addr.s6_addr[j] >> 4];
-						reverse_ip[reverse_ip_count++] = '.';
-					}
-					reverse_ip[reverse_ip_count++] = 0;
-
-					q.name = reverse_ip;
-					q.name += "ip6.arpa";
-				}
-				else
+
```
Comment (SaberUK) [inline modules/m_dns.cpp]: Extraneous whitespace here.

File: modules/m_dnsbl.cpp [hunk 6]
```diff
@@ -174,17 +173,13 @@ class ModuleDNSBL : public Module
...
 		}
 
-		const unsigned long &ip = user->ip.sa4.sin_addr.s_addr;
-		unsigned long reverse_ip = (ip << 24) | ((ip & 0xFF00) << 8) | ((ip & 0xFF0000) >> 8) | (ip >> 24);
-
-		sockaddrs reverse = user->ip;
-		reverse.sa4.sin_addr.s_addr = reverse_ip;
-
+		Anope::string reverse = user->ip.reverse();
+
```
Comment (SaberUK) [inline modules/m_dnsbl.cpp]: Here too.

File: src/sockets.cpp [hunk 7]
```diff
@@ -95,6 +95,43 @@ Anope::string sockaddrs::addr() const
...
+		{
+			unsigned long forward = sa4.sin_addr.s_addr;
+			in_addr rev;
+			rev.s_addr = forward << 24 | (forward & 0xFF00) << 8 | (forward & 0xFF0000) >> 8 | forward >> 24;
+			if (inet_ntop(AF_INET, &rev, address, sizeof(address)))
+				return address;
+			break;
+		}
+		default:
+			break;
```
Comment (SaberUK) [inline src/sockets.cpp]: This is the default behaviour.

Comment (key2peace) [inline src/sockets.cpp]: want me to change it everywhere else in sockets.cpp too ?

**Pull request merged**